src/main/java/rx/internal/operators/OperatorGroupBy.java

```diff
@@ -96,7 +93,7 @@ public GroupBySubscriber(
 
                 @Override
                 public void call() {
-                    if (WIP_FOR_UNSUBSCRIBE_UPDATER.decrementAndGet(self) == 0) {
+                    if (wipForUnsubscribe.decrementAndGet() == 0) {
```

Comment:
How should we handle that if we decide this branch should be merged? The changes I made to this class are pretty easy to re-apply if necessary